inbound.go

```diff
@@ -211,8 +211,18 @@ func (c *Connection) dispatchInbound(_ uint32, _ uint32, call *InboundCall, fram
 
 	// TODO(prashant): This is an expensive way to check for cancellation. Use a heap for timeouts.
 	go func() {
-		if <-call.mex.ctx.Done(); call.mex.ctx.Err() == context.DeadlineExceeded {
-			call.mex.inboundTimeout()
+		select {
+		case <-call.mex.ctx.Done():
+			if call.mex.ctx.Err() == context.DeadlineExceeded {
+				call.mex.inboundTimeout()
+			}
+		// If we see an error on the exchange, then we should exit immediately
+		case connErr := <-call.mex.errCh:
+			if c.log.Enabled(LogLevelDebug) {
+				c.log.Debugf("Got an error on the exchange: %v (unblock the application)", connErr)
```

Comment:
The current log message is too vague and doesn't make it clear where the error happened.

Change debug log to:
```go
c.log.Debugf("Wait for timeout/cancellation interrupted by error: %v", connErr)
```